demo-app/src/App.tsx

```diff
@@ -1,105 +1,58 @@
+import { connect } from 'react-redux';
+import { createStore } from 'redux';
+import { combinedReducers, callBack, ReduxTree, generator,
+    TreeCallBackFunction, TreeState, TreeData, Tree } from './internal';
 import 'font-awesome/css/font-awesome.min.css';
-import './App.css';
-import { Tree } from './components/Tree';
-import { generator } from './Generator';
-import { NodeProps } from './components/Node';
 
-interface AppState {
-    tree: NodeProps[];
+interface AppProps {
+    treeData?: TreeData;
+    callBack: TreeCallBackFunction;
 }
 
-class App extends React.Component<{}, AppState> {
-    private data: NodeProps[];
-
-    private actionMapper = {
-        'state.expanded': Tree.nodeExpanded,
-        'state.checked': Tree.nodeChecked,
-        'state.disabled': Tree.nodeDisabled,
-        'state.selected': Tree.nodeSelected,
-        'nodes': Tree.nodeChildren,
-        'loading': Tree.nodeLoading,
-    };
+export const store = createStore(combinedReducers, { treeData: Tree.initTree(generator())});
 
+class App extends React.Component<AppProps, {}> {
     /**
      * Constructor.
      * @param {{}} props
      */
-    constructor(props: {}) {
+    constructor(props: AppProps) {
         super(props);
-
-        this.data = Tree.initTree(generator());
-
-        console.log(Tree.nodeSearch(this.data, null, 'data-random', 'random'));
-
-        this.state = {
-            tree: this.data,
-        };
-
-        this.onDataChange = this.onDataChange.bind(this);
-        this.lazyLoad = this.lazyLoad.bind(this);
-    }
-
-    /**
-     * The callback function for changing data in the tree.
-     *
-     * @param {string} nodeId The nodeId of the node.
-     * @param {string} type The field name which changed.
-     * @param {boolean} value The new value to assign.
-     */
-    onDataChange(nodeId: string, type: string, value: boolean): void {
-        let node = Tree.nodeSelector(this.data, nodeId);
-        if ( node == null ) { return; }
-
-        if (this.actionMapper.hasOwnProperty(type)) {
-            node = this.actionMapper[type](node, value);
-            this.data = Tree.nodeUpdater(this.data, node);
-        } else {
-            // console.log(nodeId, type, value);
-        }
-
-        this.setState({tree: this.data});
-    }
-
-    /**
-     * The lazy loading function - Dummy
-     *
-     * @param {NodeProps} node The node to get children.
-     * @returns {NodeProps[]} The children.
-     */
-    lazyLoad(node: NodeProps): Promise<NodeProps[]> {
-        let isWorking = true;
-
-        return new Promise((resolve, reject) => {
-            setTimeout(() => {
-                if ( isWorking ) {
-                    resolve(generator());
-                } else {
-                    reject(new Error('Something happened.'));
-                }
-            }, 2000);
-        });
     }
 
     render() {
+        let callbacks = {
+            onDataChange: (nodeId: string, type: string, value: any) =>
```

Comment:
It's better to have this as class method because every time you re render Tree it will re assign this variable so it's better to assign on component create and reuse this callback.
```TSX
class App extends React.Component<AppProps, {}> {
  onDataChange = (nodeId: string, type: string, value: any) =>
    this.props.callBack(nodeId, type, value)
  }

 render() {
    return (
      <div className="App">
                <ReduxTree
                    hierarchicalCheck={true}
                    showCheckbox={true}
                    multiSelect={false}
                    preventDeselect={true}
                    allowReselect={true}
                    checkboxFirst={true}
                    nodeIcon={'fa fa-fw fa-circle'}
                    callbacks={ {onDataChange: this.onDataChange} }
                />
            </div>
    )
 }
}
```